var/spack/repos/builtin/packages/psblas/package.py

```diff
@@ -0,0 +1,211 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Psblas(AutotoolsPackage):
+    """PSBLAS: Parallel Sparse BLAS. A library of distributed sparse
+    linear algebra with support for GPU and Multithread acceleration.
+    Part of the PSCToolkit: Parallel Sparse Computation Toolkit."""
+
+    # Url for your package's homepage here.
+    homepage = "https://psctoolkit.github.io/"
+    git = "https://github.com/sfilippone/psblas3"
+
```

Comment:
If you add the `url` property here you can get rid of the `url` arguments to the `version` directives below AND I can confirm all -- even `3.7.0.2`'s -- sha256s.

```suggestion
    url = "https://github.com/sfilippone/psblas3/archive/refs/tags/v3.9.0-rc1.tar.gz"

```